### ISSUE TITLE
[MJAVADOC-570] --add-modules expects comma-separated arguments

### ISSUE THREAD

**Issue by jira-importer** [apache/maven-javadoc-plugin#855]
**[Gili](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=cowwoc)** opened **[MJAVADOC-570](https://issues.apache.org/jira/browse/MJAVADOC-570?redirect=false)** and commented

It's quicker for me to explain this problem that produce a minimal testcase for it. In AbstractJavadocMojo line 5079 we invoke:

```java
addArgIfNotEmpty( arguments, "--add-modules", JavadocUtil.quotedPathArgument( addModulesLine ) );
```

This gets converted to `addArgIfNotEmpty()` with `splitValue=true` which means that comma-separated values get converted to newline-separated values. 

`--add-modules` does not like this and does not consume the subsequent arguments (they probably get treated as source files instead).

We should be invoking `addArgIfNotEmpty()` with `splitValue=false` instead.

If I provide a PR for this, do you still need a testcase?

On a side-note, it would really help if they documented the expected behavior. I only ran across this by chance due to this bug report on OpenJDK: https://bugs.openjdk.java.net/browse/JDK-8165634 and when I converted newline to commas in my own project one of the error messages went away.

---

**Affects:** 3.1.0

**Remote Links:**
- [https://github.com/apache/maven-javadoc-plugin/pull/19
](https://github.com/apache/maven-javadoc-plugin/pull/19)

**Comment by jira-importer** [comment]
**[Gili](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=cowwoc)** commented

Added PR for review: https://github.com/apache/maven-javadoc-plugin/pull/19

**Comment by jira-importer** [comment]
**[Robert Scholte](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=rfscholte)** commented

Fixed in [c2681a49d246646b591281750ac0173d6f2f8834](https://gitbox.apache.org/repos/asf?p=maven-javadoc-plugin.git;a=commit;h=c2681a49d246646b591281750ac0173d6f2f8834)
Thanks for the patch!
